src/App.tsx

```diff
@@ -1,9 +1,31 @@
-import React from 'react';
+import React, { ChangeEvent, useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+
+  const visibleMovies = moviesFromServer.filter(movie => {
```

Comment:
So that `filter` is performed each time a component re-renders?

Reply:
and how can i fix it?